hs-abci-sdk/src/Tendermint/SDK/Application/AnteHandler.hs

```diff
@@ -31,32 +31,49 @@ applyAnteHandler
   -> TransactionApplication (Sem r)
 applyAnteHandler (AnteHandler ah) = ($) ah
 
+createAccountAnteHandler
+  :: Members A.AuthEffs r
+  => AnteHandler r
+createAccountAnteHandler = AnteHandler $
+  \txApplication tx@(RoutingTx Tx{..}) -> do
+    let Msg{msgAuthor} = txMsg
+    mAcnt <- A.getAccount msgAuthor
+    case mAcnt of
+      Nothing -> void $ A.createAccount msgAuthor
+      _       -> pure ()
+    txApplication tx >>= pure
+
 nonceAnteHandler
   :: Members A.AuthEffs r
   => Member (Error AppError) r
   => AnteHandler r
 nonceAnteHandler = AnteHandler $
   \txApplication tx@(RoutingTx Tx{..}) -> do
     let Msg{msgAuthor} = txMsg
-    mAcnt <- A.getAccount msgAuthor
-    account <- case mAcnt of
-      Just a@A.Account{accountNonce} -> do
-        unless (accountNonce <= txNonce) $
-          throwSDKError (NonceException accountNonce txNonce)
-        pure a
+    preMAcnt <- A.getAccount msgAuthor
+    case preMAcnt of
+      Just A.Account{accountNonce} -> do
+        let expectedNonce = accountNonce + 1
+        unless (txNonce == expectedNonce) $
+          throwSDKError (NonceException expectedNonce txNonce)
       Nothing -> do
```

Comment:
since this is no longer possible when using the `createAccountAnteH`, this should just throw an error.

Reply:
I actually tested this out yesterday... this `Nothing` case is always triggered by `checkTx` , so I ended up leaving this alone.